connections/oikotie/mapper.py

```diff
@@ -0,0 +1,358 @@
+from datetime import date, timedelta
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+from django_oikotie.enums import (
+    ApartmentType,
+    EstateType,
+    GeneralConditionLevel,
+    ModeOfHabitationType,
+    NewDevelopmentStatusChoices,
+    Site,
+)
+from django_oikotie.xml_models.apartment import (
+    Apartment,
+    Balcony,
+    CarParkingCharge,
+    City,
+    Estate,
+    FinancingFee,
+    FloorLocation,
+    GeneralCondition,
+    Lift,
+    LivingArea,
+    MaintenanceFee,
+    ModeOfHabitation,
+    NewDevelopmentStatus,
+    Picture,
+    SalesPrice,
+    Sauna,
+    ShowingDate1,
+    SiteArea,
+    UnencumberedSalesPrice,
+    WaterFee,
+    YearOfBuilding,
+)
+from django_oikotie.xml_models.housing_company import Address
+from django_oikotie.xml_models.housing_company import (
+    Apartment as HousingCompanyApartment,
+)
+from django_oikotie.xml_models.housing_company import (
+    Coordinates,
+    HousingCompany,
+    RealEstateAgent,
+)
+from typing import List, Optional
+
+from connections.elastic_models import Apartment as ElasticApartment
+from connections.utils import convert_price_from_cents_to_eur
+
+
+def map_apartment_type(elastic_apartment: ElasticApartment) -> ApartmentType:
+    project_building_type = elastic_apartment.project_building_type
+    if project_building_type == "Kerrostalo":
+        return ApartmentType.BLOCK_OF_FLATS
+    elif project_building_type == "Rivitalo":
+        return ApartmentType.ROW_HOUSE
+    elif project_building_type == "Omakotitalo":
+        return ApartmentType.HOUSE
+    else:
+        raise ValueError(
+            _("could not map the building type %s") % project_building_type
+        )
+
+
+def map_mode_of_habitation(elastic_apartment: ElasticApartment) -> ModeOfHabitation:
+    if elastic_apartment.project_holding_type == "Asumisoikeushuoneisto":
+        mode_of_habitation_type = ModeOfHabitationType.RIGHT_OF_OCCUPANCY
+    elif elastic_apartment.project_holding_type == "Osakehuoneisto":
+        mode_of_habitation_type = ModeOfHabitationType.OWNED
+    else:
+        raise ValueError(
+            _("could not map the mode of habitation %s")
+            % elastic_apartment.project_holding_type
+        )
+    return ModeOfHabitation(type=mode_of_habitation_type)
+
+
+def map_city(elastic_apartment: ElasticApartment) -> City:
+    return City(id=1, value=elastic_apartment.project_city)
```

Comment:
Without knowing the codebase this hard coded id = 1 seems a bit suspicious, but maybe it's fine?

Reply:
Yea, this is actually something I have to confirm with the Oikotie people, where they get this ID.
Now that I looked into it, this might be the "kuntanumero", which should always be Helsinki: 091.

Also now that I think about it, maybe we should keep this PR open until I send a test XML to the Oikotie people to make sure that the data looks OK.